mainwindow.cc

```diff
@@ -795,6 +795,13 @@ MainWindow::MainWindow( Config::Class & cfg_ ):
   {
     // Upgrade existing dummy tray icon into a full-functional one
 
+#if QT_VERSION >= 0x050000
+#ifdef __linux__
```

Comment:
The custom `HAVE_X11` macro is commonly used in Goldendict rather than `__linux__`.
You can also combine the two lines into one like this:
```cpp
#if defined HAVE_X11 && QT_VERSION >= QT_VERSION_CHECK( 5, 0, 0 )
```

Reply:
Please add a comment stating that this is a workaround and explaining why it is needed.